CheckSUIDPermissions.py

```diff
@@ -82,30 +91,29 @@ def check(self, pkg):
 
         files = pkg.files()
 
-        permfiles = {}
+        permfiles = set()
         # first pass, find and parse permissions.d files
-        prefix = "/etc/permissions.d/"
         for f in files:
-            if f.startswith(prefix):
+            for prefix in self._paths_to("permissions.d/"):
+                if f.startswith(prefix):
 
-                if f in pkg.ghostFiles():
-                    printError(pkg, 'polkit-ghost-file', f)
-                    continue
+                    if f in pkg.ghostFiles():
+                        printError(pkg, 'polkit-ghost-file', f)
+                        continue
 
-                bn = f[len(prefix):]
-                if bn not in _permissions_d_whitelist:
-                    printError(pkg, "permissions-unauthorized-file", f)
+                    bn = f[len(prefix):]
+                    if bn not in _permissions_d_whitelist:
+                        printError(pkg, "permissions-unauthorized-file", f)
 
-                bn = bn.split('.')[0]
-                if bn not in permfiles:
-                    permfiles[bn] = 1
+                    bn = 'permissions.d/' + bn.split('.')[0]
+                    if bn not in permfiles:
+                        permfiles.add(bn)
 
         for f in permfiles:
-            f = pkg.dirName() + "/etc/permissions.d/" + f
-            if os.path.exists(f + ".secure"):
-                self._parsefile(f + ".secure")
-            else:
-                self._parsefile(f)
+            for path in self._paths_to(f + '.secure', f):
+                if os.path.exists(path):
+                    self._parsefile(path)
+                    break
```

Comment:
You could add a comment here explicitly stating that we prefer the '.secure' variant. This intention gets a bit less clear with this change.